### DISCUSSION TITLE
format stripping feature?

### DISCUSSION THREAD

**Discussion by matanox** [Textualize/rich#3375] [Ideas] [answered]
Hi,

Any chance we could get back from Rich the string stripped of its formatting substrings? 
So that if at any point in user code we have a formatted string, we would be able to, in parallel to writing it to the terminal, write it to a file without the formatting substrings which are included in it? 

As I understand, Rich's `print` does not have a `markup:bool` attribute for this behavior, whereas its `Console` object is only meant for making output to a console, not a file, which leaves the above unsatisfiable, but maybe I'm missing an existing api for this!

The closest existing discussion may be https://github.com/Textualize/rich/discussions/2926, which is a little related but different. It might be the same as https://github.com/Textualize/rich/discussions/2032, just not in the context of using the logger class.

Admiteddly this is a bit like a logging scenario where you would not want the Rich-specific formatting markdown to appear in the log, whereas the messages are originally created in the application with Rich formatting in them. 

Thanks for Rich and thanks for your opinion!

**Comment by willmcgugan** [comment, marked as answer]
Something like this should do it.

```python
Text.from_markup("[magenta]Hello world!").plain
```
